### PR TITLE
jteng2127 proposal for pybezier

### PR DESCRIPTION
**Pull request by jteng2127**
Submit my term project proposal in `project/jteng2127/README.rst`

---
A good PR should include the following parts:

- [x] Provide a clear title
- [x] Provide meaningful descriptions in the PR and the commit logs
- [x] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [x] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [x] Exclude irrelevant files, such as binaries, text files, or dot files
- [x] Make sure tests/CI pass

### CODE REVIEW

**Review by ThreeMonth03 — COMMENTED** [review]
LGTM.

File: project/jteng2127/README.rst [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+essential to describe complex curves accurately. One approach involves
+the use of Bezier curves, which are widely applied in computer graphics
+because their smoothness and intuitive control.
+
+The goal is to use multiple third-order (or higher order) Bezier curves
+to fit the following types of curve:
+
+#. A fourth-order Bezier curve
+#. A NACA 4-digit airfoil [1]_
+#. Any arbitrary curve
```
Comment (yungyuc) [inline project/jteng2127/README.rst]: What will be the mathematic form of any arbitrary curve you are expecting to handle?

File: project/jteng2127/README.rst [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+The algorithm will be implemented in C++ for performance and efficiency,
+with Python bindings provided via pybind11.
+
+Input
+=====
+
+The fitting target can be one of the following:
+
+-  A list of points representing the curve
```
Comment (yungyuc) [inline project/jteng2127/README.rst:47]: How many data points do you plan to process in the system?

File: project/jteng2127/README.rst [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+Use Github actions for automatic building.
+
+*********************
+ Git branching model
+*********************
+
+Only `main` branch because:
+
+#. This is a term project
+#. There's only one developer :D
```
Comment (yungyuc) [inline project/jteng2127/README.rst:125]: In the beginning, yes.

File: project/jteng2127/README.rst [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+**********
+ Schedule
+**********
+
+#. 9/16~ (w3~w4)
+
+   -  Setup basic project structure and dependencies
+   -  Write some experimenting code about pybind11
+   -  Setup unit test with experimenting code
```
Comment (yungyuc) [inline project/jteng2127/README.rst:167]: Good to have unit testing planned early.

File: project/jteng2127/README.rst [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+************
+ References
+************
+
+.. _potential-solutions:
+
+Potential solutions
+===================
+
+-  `利用三階貝茲曲線逼近四階貝茲曲線__臺灣博碩士論文知識加值系統
```
Comment (yungyuc) [inline project/jteng2127/README.rst:197]: Good finding.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [x] Add an entry in `project/README.rst` pointing to the proposal.
- [x] [L27](https://github.com/yungyuc/nsdhw_24au/pull/30/files#diff-cfc04043a1f790a1e861981571bf52bc280269c403a6647dbe607eec2af9c011R27): Clarify what is your arbitrary curve form.
- [x] [L43](https://github.com/yungyuc/nsdhw_24au/pull/30/files#diff-cfc04043a1f790a1e861981571bf52bc280269c403a6647dbe607eec2af9c011R43): Make an estimate for the amount of data to be processed

File: project/jteng2127/README.rst (hunk 2 above)
Comment (jteng2127) [reply project/jteng2127/README.rst:47]: Around 20 to 200 data points during training stage, and use 2x to 5x data points during error calculation.

File: project/jteng2127/README.rst (hunk 1 above)
Comment (jteng2127) [reply project/jteng2127/README.rst]: I've updated some example for it.

**Comment by jteng2127** [comment]
@yungyuc All issues have been addressed. Please review when convenient. Thanks!

File: project/jteng2127/README.rst (hunk 1 above)
Comment (yungyuc) [reply project/jteng2127/README.rst]: Don't resolve inline comments.  It's harder to track.

File: project/jteng2127/README.rst (hunk 2 above)
Comment (yungyuc) [reply project/jteng2127/README.rst:47]: 200 are a good start.

**Review by yungyuc — APPROVED** [review]
> @yungyuc All issues have been addressed. Please review when convenient. Thanks!

LGTM.  Let me know if you want the PR to be merged.

**Comment by jteng2127** [comment]
> > @yungyuc All issues have been addressed. Please review when convenient. Thanks!
> 
> LGTM. Let me know if you want the PR to be merged.

I'm satisfied with the current proposal, please go ahead and merge this PR. Thanks!

**Pull request merged**